synapse/lib/ast.py

```diff
@@ -4161,6 +4165,83 @@ async def feedfunc():
             async for item in agen:
                 yield item
 
+class CondSetOper(Oper):
+    def __init__(self, astinfo, kids, errok=False):
+        Value.__init__(self, astinfo, kids=kids)
+        self.errok = errok
+
+    def prepare(self):
+        self.isconst = False
+        if isinstance(self.kids[0], Const):
+            self.isconst = True
+            match self.kids[0].value():
+                case "unset":
+                    self.valu = SET_UNSET
+
+    async def compute(self, runt, path):
+        if self.isconst:
+            return self.valu
+
+        valu = await self.kids[0].compute(runt, path)
+        match valu:
+            case "always":
+                return SET_ALWAYS
+            case "unset":
+                return SET_UNSET
+            case "never":
+                return SET_NEVER
+            case _:
+                mesg = f'Invalid conditional set operator ({valu}).'
+                exc = s_exc.StormRuntimeError(mesg=mesg)
+                raise self.addExcInfo(exc)
+
+class EditCondPropSet(Edit):
+
+    async def run(self, runt, genr):
+
+        if runt.readonly:
+            mesg = 'Storm runtime is in readonly mode, cannot create or edit nodes and other graph data.'
```

Comment:
Is this message copy/pasta from somewhere? Could we standardize or maybe add a `reqNotReadOnly()` to the base AST operator for consistency?

Reply:
Yeah I just split this off from EditPropSet, I'll add a helper for it

Reply:
Added a helper and cleaned up some related exceptions.